src/Related/Related.jsx

```diff
@@ -1,15 +1,46 @@
 import React from "react";
-import CardsList from './components/CardsList.jsx'
+import ListCarousel from '../Widgets/ListCarousel.jsx';
+import axios from 'axios';
+import Arrow from '../Widgets/Arrow.jsx';
+import {RiArrowRightSLine, RiArrowLeftSLine} from 'react-icons/ri';
 
 class Related extends React.Component {
   constructor(props) {
     super(props);
+    this.state = {
+      relatedProducts: []
+    }
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  async componentDidUpdate(prevProps) {
+    if (this.props.related !== prevProps.related) {
+      let promises = [];
+
+      this.props.related.forEach(id => {
+        promises.push(axios.get(`http://localhost:9000/products/${id}`).then(response => response.data));
+      });
+
+      await Promise.all(promises)
```

Comment:
Very cool and intelligent use of Promise.all. I like it. Prevents some async from getting all, well, asynchronous.